src/samVisualize/graphGeneration/cfnTemplateGraphGenerator.ts

```diff
@@ -0,0 +1,150 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Graph, GraphObject } from './graph'
+import { yamlParse } from 'yaml-cfn'
+import { LinkTypes } from '../samVisualizeTypes'
+import * as _ from 'lodash'
+import { getLogger } from '../../shared/logger/logger'
+
+/**
+ * @param inputYaml A string representing a YAML template
+ * @returns A JavaScript object corresponding to the given YAML string, or undefined if the input is not valid YAML
+ */
+function yamlStringToObject(inputYaml: string): Record<string, any> | undefined {
+    try {
+        return yamlParse(inputYaml)
+    } catch (err) {
+        getLogger().error(
+            `SAM Visualize: Failed to load template "${inputYaml.substr(
+                0,
+                25
```

Comment:
Why 25?

Reply:
It was discussed [here](https://github.com/aws/aws-toolkit-vscode/pull/1873#discussion_r670795818). 25 characters is arbitrary, I just wanted to give a little preview of the template that was failing without printing the entire string. I'll leave a comment to explain.